docs/hugo/content/contributing/add-a-new-code-generated-resource/before-you-begin.md

```diff
@@ -0,0 +1,98 @@
+---
+title: Before you begin adding a new code-generated resource
+linktitle: Before you begin
+weight: 10
+---
+
+## Determine a resource to add
+
+The first step in adding a new code-generated resource is to determine which resource you want to add. 
+
+Any ARM resource can be generated. If you're not sure if the resource you are interested in is an ARM resource, check if it is defined in a `resource-manager` folder in the [Azure REST API specs](https://github.com/Azure/azure-rest-api-specs/tree/main/specification) repo.
+
+If it is, it's an ARM resource.
+
+To unambigously identify the resource you want to add, you need to know its _**group**_, _**version**_, and _**kind**_ (aka GVK).
+
+We work out these by reviewing the [Azure REST API specs](https://github.com/Azure/azure-rest-api-specs/tree/main/specification) repo. We will walk through an example of adding Azure Synapse Workspace.
+
+### Kind
+
+Also known as the _name_ of the resource, you usually know this going in. In our example above, the name of the resource is `workspace`. 
+
+If you're not sure, one approach is to look in the Swagger/OpenAPI specification file for the service and find the documented PUT for the resource you're interested in. The resource name will be the
+[second to last section of the URL](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/synapse/resource-manager/Microsoft.Synapse/stable/2021-06-01/workspace.json#L71).
```

Comment:
"second to last section" isn't very clear to me. Isn't it the json filename at the end of the URL?